src/charm.py

```diff
@@ -917,6 +1124,18 @@ def _set_ca_certificate_secret_in_peer_relation(
         peer_relation.data[self.app].update({CA_CERTIFICATE_JUJU_SECRET_KEY: juju_secret.id})  # type: ignore[union-attr]  # noqa: E501
         logger.info("Vault CA certificate secret set in peer relation")
 
+    def _remove_ca_certificate_secret_in_peer_relation(self) -> None:
+        """Removes the vault CA certificate secret in the peer relation.
+
+        Only the key from the relation is removed. The actual juju secret is unmodified.
+        """
+        if not self._is_peer_relation_created():
+            raise RuntimeError("Peer relation not created")
+        peer_relation = self.model.get_relation(PEER_RELATION_NAME)
+        if peer_relation:
+            peer_relation.data[self.app].pop(CA_CERTIFICATE_JUJU_SECRET_KEY)
+        logger.info("Vault CA certificate secret removed from peer relation")
```

Comment:
The log should go with the `pop`. Here it's possible that we log and that `xyz was removed` but that nothing was done.

Reply:
Another oops. fixed.